Students/tylerkobienia/PART4LAB1.2/PART4LAB1.2.ino

```diff
@@ -0,0 +1,160 @@
+//Part4 Lab1
+
+// pin ins
+const int redPin = 9;        
+const int greenPin = 10;     
+const int bluePin = 11;      
+const int buttonPin = 3;    
+
+// mode constants
+const int OFF = 1;            
+const int BLINK_WHITE = 2;    
+const int FADE_PURPLE = 3;    
+const int FADE_RED_BLUE = 4;  
+const int FLASH_BLUE_GREEN = 5;
+
+// variable to store the current mode
+int currentMode = OFF;
+
+// variables to handle button state
+int lastButtonState = HIGH;    // variable to store the last button state
+int buttonState;               // variable to store the current button state
+
+// setup function runs once at the beginning
+void setup() {
+  // set pin modes for RGB and button
+  pinMode(redPin, OUTPUT);     // set red led pin as OUTPUT
+  pinMode(greenPin, OUTPUT);   // set green led pin as OUTPUT
+  pinMode(bluePin, OUTPUT);    // set blue led pin as OUTPUT
+  pinMode(buttonPin, INPUT_PULLUP);  // set button pin as INPUT_PULLUP
+
+  // initialize serial communication for debugging
+  Serial.begin(9600);
+}
+
+// main loop function runs repeatedly
+void loop() {
+  // Read the state of the button
+  buttonState = digitalRead(buttonPin);
+
+  // check if the button is pressed (LOW) and was not pressed in the last loop
+  if (buttonState == LOW && lastButtonState == HIGH) {
+    // change to the next mode
+    changeMode();
+    // add a delay to debounce the button press
+    delay(1000);
```

Comment:
That is a very long delay (1s). Did that have any negative effects on your design?